solr/src/main/java/org/apache/metamodel/solr/SolrDataContext.java

```diff
@@ -0,0 +1,396 @@
+
+            Map<String, Object> schemaMap = null;
+
+            if (outerMap != null) {
+                schemaMap = (Map) outerMap.get("schema");
+            }
+
+            if (schemaMap != null) {
+                List<Map<String, String>> fieldMapList = (List) schemaMap
+                        .get("fields");
+                List<String> columnNames = new ArrayList<String>();
+                List<ColumnType> columnTypes = new ArrayList<ColumnType>();
+
+                for (int i = 0; i < fieldMapList.size(); i++) {
+                    Map<String, String> fieldMap = fieldMapList.get(i);
+
+                    columnNames.add(fieldMap.get("name"));
+                    columnTypes.add(ColumnType.STRING);
+                }
+
+                String[] columnNamesArr = new String[columnNames.size()];
+                columnNamesArr = columnNames.toArray(columnNamesArr);
+
+                ColumnType[] columnTypesArr = new ColumnType[columnTypes.size()];
+                columnTypesArr = columnTypes.toArray(columnTypesArr);
+
+                tableDef = new SimpleTableDef(indexName, columnNamesArr,
+                        columnTypesArr);
+            }
+        } catch (IOException e) {
+            logger.error("Failed to parse schema", "", e);
+        }
+
+        return tableDef;
+    }
+
+    @Override
+    protected Schema getMainSchema() throws MetaModelException {
+        try {
+            final MutableSchema theSchema = new MutableSchema(
+                    getMainSchemaName());
+
+            final MutableTable table = tableDef.toTable().setSchema(theSchema);
+            theSchema.addTable(table);
+
+            return theSchema;
+        } catch (Exception e) {
+            throw new MetaModelException("Schema retrieval failed " + e);
+        }
+    }
+
+    @Override
+    protected Number executeCountQuery(Table table,
+            List<FilterItem> whereItems, boolean functionApproximationAllowed) {
+        String queryStr = "*:*";
+
+        if (whereItems != null) {
+            for (int i = 0; i < whereItems.size(); i++) {
+                FilterItem whereItem = whereItems.get(i);
+                queryStr = whereItem.toString();
+            }
+        }
+
+        QueryResponse response = selectRows(table, null, queryStr, 0, 0);
+        return response.getResults().getNumFound();
+    }
+
+    @Override
+    public DataSet executeQuery(Query q) throws MetaModelException {
+        String queryStr = "";
+        QueryResponse response = null;
+
+        Schema schema = getMainSchema();
+        Table table = schema.getTable(0);
+        Column[] columns = table.getColumns();
+
+        boolean selectAll = true;
+        List<SelectItem> selectItems = q.getSelectClause().getItems();
+
+        for (SelectItem selectItem : selectItems) {
+            if (selectItem.getFunction() != null
+                    || selectItem.getColumn() == null) {
```

Comment:
This loop seems to check for any SelectItem that has a function and no column. But you update a variable called "selectAll". If your intent is to find out if this is a SELECT COUNT(*) FROM [table] kind of query, then there is a much safer way to implement it - to overrid the executeCountQuery(...) method of the superclass.